### PR TITLE
chore: get rid of useDualWorkflow

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

https://github.com/dust-tt/tasks/issues/763 (1/2)

- remove all logic associated with "useDualWorkflow"
- make the field nullable
- stop filing it

## Risk

Logic isn't used anymore.

## Deploy Plan

1. run migration (prodbox)
2. deploy code
3. (other PR) drop the column

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/client.ts [hunk 1]
```diff
@@ -60,11 +58,11 @@ export async function launchNotionSyncWorkflow(
         connectorId,
         startFromTs,
         forceResync,
-        garbageCollectionMode: useDualWorkflow ? "never" : "auto",
+        garbageCollectionMode: "never",
       },
     ],
     taskQueue: QUEUE_NAME,
-    workflowId: getNotionWorkflowId(dataSourceConfig),
+    workflowId: getNotionWorkflowId(dataSourceConfig, "never"),
```
Comment (spolu) [inline connectors/src/connectors/notion/temporal/client.ts:65]: that's change no?

**Review by spolu — APPROVED** [review]

File: connectors/src/connectors/notion/temporal/client.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/notion/temporal/client.ts:65]: (a change of behavior I mean)

File: types/src/connectors/notion.ts [hunk 2]
```diff
@@ -8,11 +8,11 @@ import * as t from "io-ts";
 export type PageObjectProperties = PageObjectResponse["properties"];
 export type PropertyKeys = keyof PageObjectProperties;
 export type PropertyTypes = PageObjectProperties[PropertyKeys]["type"];
-export type NotionGarbageCollectionMode = "always" | "auto" | "never";
+export type NotionGarbageCollectionMode = "always" | "never";
 
 export function getNotionWorkflowId(
   dataSourceInfo: { workspaceId: string; dataSourceName: string },
-  gargbageCollectionMode: NotionGarbageCollectionMode = "auto"
+  gargbageCollectionMode: NotionGarbageCollectionMode
```
Comment (flvndvd) [inline types/src/connectors/notion.ts:15]: Can we use something else than `never` and `always` here, maybe a regular boolean, like `isGarbageCollectorWorkflow`?

File: connectors/src/connectors/notion/temporal/client.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/notion/temporal/client.ts:65]: ➕

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/client.ts:65]: nope

File: types/src/connectors/notion.ts (hunk 2 above)
Comment (fontanierh) [reply types/src/connectors/notion.ts:15]: WHy ?

File: connectors/src/connectors/notion/temporal/client.ts (hunk 1 above)
Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/client.ts:65]: "Never" and "auto" generate the same workflow id

File: types/src/connectors/notion.ts (hunk 2 above)
Comment (flvndvd) [reply types/src/connectors/notion.ts:15]: Cause it's not idiomatic! When you see the call sites, you can't really tell how this parameter influences the  workflow id.

Comment (fontanierh) [reply types/src/connectors/notion.ts:15]: I don't think idiomatic is the right word here, but agree it's not super explicit on call sites.

Comment (flvndvd) [reply types/src/connectors/notion.ts:15]: Thank you 🙇 .

**Pull request merged**